content/docs/context.md

```diff
@@ -4,220 +4,110 @@ title: Context
 
-If `contextTypes` is defined within a component, the following [lifecycle methods](/docs/react-component.html#the-component-lifecycle) will receive an additional parameter, the `context` object:
+### `Consumer`
 
-- [`constructor(props, context)`](/docs/react-component.html#constructor)
-- [`componentWillReceiveProps(nextProps, nextContext)`](/docs/react-component.html#componentwillreceiveprops)
-- [`shouldComponentUpdate(nextProps, nextState, nextContext)`](/docs/react-component.html#shouldcomponentupdate)
-- [`componentWillUpdate(nextProps, nextState, nextContext)`](/docs/react-component.html#componentwillupdate)
+```js
+<Consumer>
+  {value => /* render something based on the context value */}
+</Consumer>
+```
 
-> Note:
->
-> As of React 16, `componentDidUpdate` no longer receives `prevContext`.
+A React component that subscribes to context changes.
 
-## Referencing Context in Stateless Functional Components
+Requires a [function as a child](/docs/render-props.html#using-props-other-than-render). The function receives the current context value and returns a React node. All consumers are re-rendered whenever the Provider value changes. Changes are determined by comparing the new and old values using [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is).
 
-Stateless functional components are also able to reference `context` if `contextTypes` is defined as a property of the function. The following code shows a `Button` component written as a stateless functional component.
+> Note
+> 
+> For more information about this pattern, see [render props](/docs/render-props.html).
 
-```javascript
-import PropTypes from 'prop-types';
+## Examples
 
-const Button = ({children}, context) =>
-  <button style={{background: context.color}}>
-    {children}
-  </button>;
+### Dynamic Context
 
-Button.contextTypes = {color: PropTypes.string};
-```
+A more complex example with dynamic values for the theme:
 
-## Updating Context
+**theme-context.js**
+`embed:context/theme-detailed-theme-context.js`
 
-Don't do it.
+**themed-button.js**
+`embed:context/theme-detailed-themed-button.js`
 
-React has an API to update context, but it is fundamentally broken and you should not use it.
+**app.js**
+`embed:context/theme-detailed-app.js`
 
-The `getChildContext` function will be called when the state or props changes. In order to update data in the context, trigger a local state update with `this.setState`. This will trigger a new context and changes will be received by the children.
+## Consuming Multiple Contexts
 
-```javascript
-import PropTypes from 'prop-types';
+`embed:context/multiple-contexts.js`
 
-class MediaQuery extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {type:'desktop'};
-  }
+## Accessing Context in Lifecycle Methods
 
```

Comment:
Again, needs an intro before the example. Explain that this is a relatively common use case. And that instead of adding context to every lifecycle method, you just need to pass it as a prop, and then work with it just like you'd normally work with a prop.